homeassistant/components/transmission/const.py

```diff
@@ -1,10 +1,34 @@
 """Constants for the Transmission Bittorent Client component."""
-
 DOMAIN = "transmission"
 
 SWITCH_TYPES = {"on_off": "Switch", "turtle_mode": "Turtle Mode"}
 
+ORDER_AGE = "age"
+ORDER_AGE_DESC = "age_desc"
+ORDER_ID = "id"
+ORDER_ID_DESC = "id_desc"
+ORDER_RATIO = "ratio"
+ORDER_RATIO_DESC = "ratio_desc"
+
+SUPPORTED_ORDER_MODES = {
+    ORDER_AGE: lambda torrents: sorted(
+        torrents, key=lambda t: t.addedDate, reverse=True
+    ),
+    ORDER_AGE_DESC: lambda torrents: sorted(torrents, key=lambda t: t.addedDate),
+    ORDER_ID: lambda torrents: sorted(torrents, key=lambda t: t.id),
+    ORDER_ID_DESC: lambda torrents: sorted(torrents, key=lambda t: t.id, reverse=True),
+    ORDER_RATIO: lambda torrents: sorted(torrents, key=lambda t: t.ratio),
+    ORDER_RATIO_DESC: lambda torrents: sorted(
+        torrents, key=lambda t: t.ratio, reverse=True
+    ),
+}
+
+CONF_LIMIT = "limit"
+CONF_ORDER = "order"
+
```

Comment:
At first I thought _DESC was "description" and was confused.  I see it is blank for ascending and _DESC for descending.  Consider ORDER_NEWEST / ORDER_OLDEST.  For ratio I am not sure - ORDER_RATIO_BEST / ORDER_RATIO_WORST feels bad.

If you'd like to keep it consistent I think that it would benefit from adding _ASC to the ascending ones.

Is there a reason someone would want to sort by ID ascending vs descending?

Reply:
Yeah, naming is a bit confusing, but I had to start with something, I'll try to come up with something more obvious. BTW, do you know how to translate the order options on the config flow form?